Admin/AbstractAdmin.php

```diff
@@ -1971,50 +1965,32 @@ public function buildBreadcrumbs($action, MenuItemInterface $menu = null)
             return $this->breadcrumbs[$action];
         }
 
-        if (!$menu) {
-            $menu = $this->menuFactory->createItem('root');
+        return $this->breadcrumbs[$action] = $this->getBreadcrumbsBuilder()
+            ->buildBreadcrumbs($this, $action, $menu);
+    }
 
-            $menu = $menu->addChild(
-                $this->trans($this->getLabelTranslatorStrategy()->getLabel('dashboard', 'breadcrumb', 'link'), array(), 'SonataAdminBundle'),
-                array('uri' => $this->routeGenerator->generate('sonata_admin_dashboard'))
-            );
+    /**
+     * @return BreadcrumbsBuilderInterface
+     */
+    final public function getBreadcrumbsBuilder()
```

Comment:
Did you read [this](http://ocramius.github.io/blog/when-to-declare-classes-final/) ? I think it is a good way to get information about how the lib is used and building the best abstraction. In this particular instance, I can't imagine why people would want to change this method, so let's make it `final`.